diff --git a/src/common/types/interval.cpp b/src/common/types/interval.cpp
index 5bc9c1aa2d..8adb2b8874 100644
--- a/src/common/types/interval.cpp
+++ b/src/common/types/interval.cpp
@@ -45,7 +45,7 @@ bool IntervalTryAddition(T &target, int64_t input, int64_t multiplier, string *e
 	}
 	T addition_base;
 	if (!TryCast::Operation<int64_t, T>(addition, addition_base)) {
-		AssignInvalidInputErrorOrThrow(CastExceptionText<int64_t, T>(addition), error_message);
+		AssignInvalidInputErrorOrThrow(CastExceptionText<int64_t>(addition, TypeIdInfo::CreateObject<T>), error_message);
 		return false;
 	}
 	if (!TryAddOperator::Operation<T, T, T>(target, addition_base, target)) {
@@ -57,7 +57,7 @@ bool IntervalTryAddition(T &target, int64_t input, int64_t multiplier, string *e
 		//	This is always in range
 		addition = (fraction * multiplier) / Interval::MICROS_PER_SEC;
 		if (!TryCast::Operation<int64_t, T>(addition, addition_base)) {
-			AssignInvalidInputErrorOrThrow(CastExceptionText<int64_t, T>(addition), error_message);
+			AssignInvalidInputErrorOrThrow(CastExceptionText<int64_t>(addition, TypeIdInfo::CreateObject<T>), error_message);
 			return false;
 		}
 		if (!TryAddOperator::Operation<T, T, T>(target, addition_base, target)) {
@@ -142,7 +142,8 @@ interval_parse_number:
 		}
 		string_t nr_string(str + start_pos, UnsafeNumericCast<uint32_t>(pos - start_pos));
 		if (!TryCast::Operation<string_t, int64_t>(nr_string, number)) {
-			AssignInvalidInputErrorOrThrow(CastExceptionText<string_t, int64_t>(nr_string), error_message);
+			AssignInvalidInputErrorOrThrow(CastExceptionText<string_t>(nr_string, TypeIdInfo::CreateObject<int64_t>),
+			                               error_message);
 			return false;
 		}
 		fraction = 0;
